### ISSUE TITLE
Load a database from the server Failed  Error: file is encrypted or is not a database

### ISSUE THREAD

**Issue by wuxingjun** [kripken/sql.js#185]
Load a database from the server Failed  Error: file is encrypted or is not a database
Here are my codes.   test.db is in the root directory of my website
var xhr = new XMLHttpRequest();
xhr.open('GET', 'test.db', true);
xhr.responseType = 'arraybuffer';

xhr.onload = function(e) {
  var uInt8Array = new Uint8Array(this.response);
  var db = new SQL.Database(uInt8Array);
  var contents = db.exec("SELECT * FROM testtable");
};
xhr.send();

**Comment by lovasoa** [comment]
Maybe test.db is encrypted or is not a database? ;)

2 января 2017 г. 16:02:32 GMT+01:00, wuxingjun <notifications@github.com> пишет:
>Load a database from the server Failed  Error: file is encrypted or is
>not a database
>Here are my codes.   test.db is in the root directory of my website
>var xhr = new XMLHttpRequest();
>xhr.open('GET', 'test.db', true);
>xhr.responseType = 'arraybuffer';
>
>xhr.onload = function(e) {
>  var uInt8Array = new Uint8Array(this.response);
>  var db = new SQL.Database(uInt8Array);
>  var contents = db.exec("SELECT * FROM testtable");
>};
>xhr.send();
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/kripken/sql.js/issues/185

-- 
Простите за краткость, создано в K-9 Mail.

**Comment by wuxingjun** [comment]
I ensure test.db  is a database and not encrypted.

I can get data by upload sqlite file,but not from server
<input id="uploadInput" type="file" onchange="getfile()">
  function getfile() {
            var files = document.getElementById("uploadInput").files;
            var file = files[0];
            var fr = new FileReader();
            fr.onload = function () {
                var Uints = new Uint8Array(fr.result);
                var db = new window.SQL.Database(Uints);
                var res = db.exec("SELECT * FROM testtable");
                for (i = 0; i < res[0].values.length; i++) {
                    console.log(res[0].values[i] + ' ');
                }
            }
            fr.readAsArrayBuffer(file);
        }

**Comment by lovasoa** [comment]
Is that your raw html file ? If yes, you are missing a `<script>` tag around your code.

**Comment by wuxingjun** [comment]
Here are my test codes and DB file
![image](https://cloud.githubusercontent.com/assets/24877354/21598813/101cedd8-d19e-11e6-9204-e16e32291bbc.png)
index.html
![image](https://cloud.githubusercontent.com/assets/24877354/21598834/6c31a49c-d19e-11e6-9d09-8d0f9bdfdf62.png)
![image](https://cloud.githubusercontent.com/assets/24877354/21598856/b87ed3ce-d19e-11e6-8ddf-9c4bdf467ac5.png)

**Comment by lovasoa** [comment]
This is not what I asked you. What are the URLs of your files? Display index.html in your browser and give me the url. Open the database in your browser and give me the url. 

3 января 2017 г. 3:26:34 GMT+01:00, wuxingjun <notifications@github.com> пишет:
>
>Here are my test codes and DB file
>![image](https://cloud.githubusercontent.com/assets/24877354/21598813/101cedd8-d19e-11e6-9204-e16e32291bbc.png)
>index.html
>![image](https://cloud.githubusercontent.com/assets/24877354/21598834/6c31a49c-d19e-11e6-9d09-8d0f9bdfdf62.png)
>![image](https://cloud.githubusercontent.com/assets/24877354/21598856/b87ed3ce-d19e-11e6-8ddf-9c4bdf467ac5.png)
>
>
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/kripken/sql.js/issues/185#issuecomment-270043209

-- 
Простите за краткость, создано в K-9 Mail.

**Comment by wuxingjun** [comment]
I've found the reason.Permission Issue

**Comment by wuxingjun** [comment]
That means Anyone can download the db file from website,right?

**Comment by lovasoa** [comment]
Of course. If you want to read the database from the browser, then the database has to be accessible from the browser. 

5 января 2017 г. 2:29:23 GMT+01:00, wuxingjun <notifications@github.com> пишет:
>That means Anyone can download the db file from website,right?
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/kripken/sql.js/issues/185#issuecomment-270538688

-- 
Простите за краткость, создано в K-9 Mail.

**Comment by NameNull** [comment]
how to solve this problem?
